src/link_manager/LinkManagerStateMachineActiveStandby.cpp

```diff
@@ -237,13 +237,19 @@ void ActiveStandbyStateMachine::setLabel(Label label) {
 };
 
 //
-// ---> enterLinkProberState(CompositeState &nextState, link_prober::LinkProberState::Label label);
+// ---> enterLinkProberState(CompositeState &nextState, link_prober::LinkProberState::Label label, bool forceReset);
 //
 // force LinkProberState to switch state
 //
-void ActiveStandbyStateMachine::enterLinkProberState(CompositeState &nextState, link_prober::LinkProberState::Label label)
+void ActiveStandbyStateMachine::enterLinkProberState(CompositeState &nextState, link_prober::LinkProberState::Label label, bool forceReset)
 {
     mLinkProberStateMachinePtr->enterState(label);
+
+    if (forceReset && ps(nextState) == label) {
+        // only need to reset the link prober state if the state remains the same
+        mLinkProberStateMachinePtr->resetCurrentState();
+    }
```

Comment:
@lolyu - if I understand the issue correctly, the extra toggle is trigged because the event counters are not reset when we use `enterLinkProberState` to force into a state? 

If so, why don't we just reset every time we call `enterLinkProberState`? We should still get 3 continuous heartbeat missing events before entering `LinkProberUnknon`.

Reply:
The issue is only observed during link `up`, this is to limit the change to fix this issue. We don't know what else could be caused if we reset the link prober state every time we call `enterLinkProberState`.